modules/ROOT/pages/patterns/fixed-length-patterns.adoc

```diff
@@ -0,0 +1,223 @@
+A `WHERE` clause can be used for more general predicates:
+
+[source, role=noheader]
+----
+-[r WHERE time() + duration(r.duration) < time('22:00') ]->
+----
+
+See the xref:patterns/reference.adoc#relationship-patterns[relationship patterns] reference section for more details.
+
+[[path-patterns]]
+== Path patterns
+
+Any valid path starts and ends with a node, with relationships between each node (if there is more than one node).
+Path patterns have the same restrictions, and for all valid path patterns the following are true:
+
+* They have at least one node pattern.
+* They begin and end with a node pattern.
+* They alternate between nodes and relationships.
+
+These are all valid path patterns:
+
+[source, role=noheader]
+----
+()
+----
+[source, role=noheader]
+----
+(s)--(e)
+----
+[source, role=noheader]
+----
+(:Station)--()<--(m WHERE m.departs > time('12:00'))-->()-[:NEXT]->(n)
+----
+
+These are invalid path patterns:
+
+[source, role=noheader]
+----
+-->
+----
+[source, role=noheader]
+----
+()-->
+----
+[source, role=noheader]
+----
+()-->-->()
+----
+
+[[path-pattern-matching]]
+== Path pattern matching
+
+This section contains an example of matching a path pattern to paths in a property graph.
+
+It uses the following graph:
+
+image::path_pattern_example_graph.svg[width="600",role="middle"]
+
+To recreate the graph, run the following query against an empty Neo4j database:
+
+[source,cypher, role=test-setup]
+----
+CREATE (pmr:Station {name: 'Peckham Rye'}),
+  (dmk:Station {name: 'Denmark Hill'}),
+  (vic:Station {name: 'London Victoria'}),
+  (clp:Station {name: 'Clapham High Street'}),
+  (eph:Station {name: 'Elephant & Castle'}),
+  (vic)<-[:CALLS_AT]-(s1:Stop {departs: time('11:55')}),
+  (dmk)<-[:CALLS_AT]-(s2:Stop {departs: time('11:44')})-[:NEXT]->(s1),
+  (pmr)<-[:CALLS_AT]-(s3:Stop {departs: time('11:40')})-[:NEXT]->(s2),
+  (clp)<-[:CALLS_AT]-(s4:Stop {departs: time('11:41')}),
+  (dmk)<-[:CALLS_AT]-(s5:Stop {departs: time('11:37')})-[:NEXT]->(s4),
+  (pmr)<-[:CALLS_AT]-(s6:Stop {departs: time('11:33')})-[:NEXT]->(s5),
+  (eph)<-[:CALLS_AT]-(s7:Stop {departs: time('11:54')}),
+  (dmk)<-[:CALLS_AT]-(s8:Stop {departs: time('11:47')})-[:NEXT]->(s7),
+  (pmr)<-[:CALLS_AT]-(s9:Stop {departs: time('11:44')})-[:NEXT]->(s8)
+----
+
+
+The graph contains a number of train `Stations` and `Stops`.
+A `Stop` represents  the arrival and departure of a train that `CALLS_AT` a `Station`.
+Each `Stop` forms part of a sequence of `Stops` connected by relationships with the type `NEXT`, representing the order of calling points made by a train service.
+
+The graph shows three chains of `Stops` that represent different train services.
+Each of these services calls at the `Station` with the name `Denmark Hill`.
+
+To return all `Stops` that call at the `Station` `Denmark Hill`, the following _motif_ is used (the term motif is used to describe the pattern looked for in the graph):
+
+image::path_pattern_motif.svg[width="600",role="middle"]
```

Comment:
This file was apparently already there but maybe better late than never:
I just noticed that this looks a bit blurry on my screen: It seems like this is a png embedded into an svg. Also the font in the image does not match the font in the text. ("Public Sans" and "Roboto Mono" on my computer)

Reply:
Shock revelation: many of these diagrams that have been composed on Google Docs are fake SVGs. As this one is a bit blurry, I have properly recreated it as an SVG.